src/main/java/bio/terra/service/iam/sam/SamIam.java

```diff
@@ -526,13 +527,13 @@ public static DataRepoException convertSAMExToDataRepoEx(final ApiException samE
         logger.warn("SAM client exception message: {}", samEx.getMessage());
         logger.warn("SAM client exception details: {}", samEx.getResponseBody());
 
-        // Sometimes the sam message is buried one level down inside of the error report object.
+        // Sometimes the sam message is buried several levels down inside of the error report object.
         // If we find an empty message then we try to deserialize the error report and use that message.
         String message = samEx.getMessage();
         if (StringUtils.isEmpty(message)) {
             try {
                 ErrorReport errorReport = objectMapper.readValue(samEx.getResponseBody(), ErrorReport.class);
-                message = errorReport.getMessage();
+                message = extractErrorMessage(errorReport);
```

Comment:
This is only called if the message in `ApiException` is empty. Are these cases where the underlying causes are informative even if the message is non-empty?

Reply:
Yeah, the case in point: if you try to create a resource in sam with an invalid email you get the following back from Sam:
```{
  "causes": [
    {
      "causes": [
        {
          "causes": [
            {
              "causes": [],
              "message": "Invalid member email: harry.potterxyzsaasasassa@test.firecloud.org",
              "source": "sam",
              "stackTrace": []
            }
          ],
          "message": "You have specified at least one invalid member email",
          "source": "sam",
          "stackTrace": []
        }
      ],
      "message": "You have specified an invalid policy",
      "source": "sam",
      "stackTrace": []
    }
  ],
  "message": "Cannot create resource",
  "source": "sam",
  "stackTrace": [],
  "statusCode": 400
}```

And in this case, it doesn't appear to map that to `samEx.getMessage()`